onnxruntime/core/framework/path_lib.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "path_lib.h"
+#include "core/common/status.h"
+#include "core/common/common.h"
+#include <assert.h>
+#ifdef _WIN32
+#include <shlwapi.h>
+#else
+#include <libgen.h>
+#endif
+
+#ifdef _WIN32
+namespace onnxruntime {
+namespace {
+//starting from pszPath_end-1, backsearch the first char that is not L'/'
+inline void backsearch(PWSTR pszPath, PWSTR& pszPath_end) {
```

Comment:
>inline [](start = 0, length = 6)

I don't think 'inline' does anything when it's not used in a header file. 

nit: BackSearch